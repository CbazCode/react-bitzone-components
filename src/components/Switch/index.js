import {Ball, InputSwitch, LabelSwitch, SwitchContainer} from "./Switch";

const Switch = ({id, isDisabled,isChecked, defaultCheked, isRequired, name, size, bgColor}) => {
    console.log(defaultCheked);
    return (
        <SwitchContainer defaultCheked ={defaultCheked}>
            <InputSwitch className="checkbox" id={id} isDisabled={isDisabled} isChecked ={isChecked} isRequired={isRequired} name={name}/>
            <LabelSwitch className="label" htmlFor={id} size={size} bgColor={bgColor}>
                <Ball className="ball" defaultCheked ={defaultCheked} size={size}/>
            </LabelSwitch>
        </SwitchContainer>  
        
    );
}
 
export default Switch;