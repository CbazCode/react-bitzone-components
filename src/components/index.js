export { default as Button } from './Button/Button';
export { default as Drawer } from './Drawer/Drawer';
export { default as Input } from './Input/Input';
export { default as Modal } from './Modal/Modal';
export { default as Skeleton } from './Skeleton/Skeleton';
export { default as Slider } from './Slider/Slider';
export { default as Spinner } from './Spinner/Spinner';
export { default as Switch } from './Switch/Switch';
