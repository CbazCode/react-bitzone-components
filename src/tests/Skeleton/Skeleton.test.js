import { Skeleton } from '../../components';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

describe('Test en <Skeleton />', () => {
  test('componente <Skeleton /> debe renderizarse correctamente', () => {
    const { container } = render(<Skeleton />);
    expect(container.firstChild).toMatchSnapshot();
  });
  test('componente debe funcionar con parametros por defecto', () => {
    const skeleton = renderer.create(<Skeleton />).toJSON();
    expect(skeleton).toHaveStyleRule('width', '100%');
    expect(skeleton).toHaveStyleRule('height', 'auto');
    expect(skeleton).toHaveStyleRule('margin-top', '5px');
    expect(skeleton).toHaveStyleRule('margin-right', '5px');
    expect(skeleton).toHaveStyleRule('margin-bottom', '5px');
    expect(skeleton).toHaveStyleRule('margin-left', '5px');
    expect(skeleton).toHaveStyleRule('background-color', '#cacaca');
    expect(skeleton).toHaveStyleRule('border-radius', '4px');
    expect(skeleton).toHaveStyleRule('color', 'transparent');
    expect(skeleton).toHaveStyleRule('user-select', 'none');
    expect(skeleton).toHaveStyleRule('animation', 'dKHzAH 3s linear infinite');
  });
  test('componente debe funcionar con propiedades - default', () => {
    const skeleton = renderer
      .create(<Skeleton height="20px" mt="6px" mr="6px" mb="6px" ml="6px" />)
      .toJSON();
    expect(skeleton).toHaveStyleRule('width', '100%');
    expect(skeleton).toHaveStyleRule('height', '20px');
    expect(skeleton).toHaveStyleRule('margin-top', '6px');
    expect(skeleton).toHaveStyleRule('margin-right', '6px');
    expect(skeleton).toHaveStyleRule('margin-bottom', '6px');
    expect(skeleton).toHaveStyleRule('margin-left', '6px');
    expect(skeleton).toHaveStyleRule('background-color', '#cacaca');
    expect(skeleton).toHaveStyleRule('border-radius', '4px');
    expect(skeleton).toHaveStyleRule('color', 'transparent');
    expect(skeleton).toHaveStyleRule('user-select', 'none');
    expect(skeleton).toHaveStyleRule('animation', 'dKHzAH 3s linear infinite');
  });
  test('componente debe funcionar con type circle', () => {
    const skeleton = renderer.create(<Skeleton type="circle" />).toJSON();
    expect(skeleton).toHaveStyleRule('width', 'calc(5px*10)');
    expect(skeleton).toHaveStyleRule('height', 'calc(5px*10)');
    expect(skeleton).toHaveStyleRule('margin-top', '5px');
    expect(skeleton).toHaveStyleRule('margin-right', '5px');
    expect(skeleton).toHaveStyleRule('margin-bottom', '5px');
    expect(skeleton).toHaveStyleRule('margin-left', '5px');
    expect(skeleton).toHaveStyleRule('background-color', '#cacaca');
    expect(skeleton).toHaveStyleRule('border-radius', '50%');
    expect(skeleton).toHaveStyleRule('color', 'transparent');
    expect(skeleton).toHaveStyleRule('animation', 'dKHzAH 3s linear infinite');
  });
  test('componente debe funcionar con type circle y size', () => {
    const skeleton = renderer.create(<Skeleton type="circle" size={10} />).toJSON();
    expect(skeleton).toHaveStyleRule('width', 'calc(10px*10)');
    expect(skeleton).toHaveStyleRule('height', 'calc(10px*10)');
    expect(skeleton).toHaveStyleRule('margin-top', '5px');
    expect(skeleton).toHaveStyleRule('margin-right', '5px');
    expect(skeleton).toHaveStyleRule('margin-bottom', '5px');
    expect(skeleton).toHaveStyleRule('margin-left', '5px');
    expect(skeleton).toHaveStyleRule('background-color', '#cacaca');
    expect(skeleton).toHaveStyleRule('border-radius', '50%');
    expect(skeleton).toHaveStyleRule('color', 'transparent');
    expect(skeleton).toHaveStyleRule('animation', 'dKHzAH 3s linear infinite');
  });

  test('componente debe mostrar contenido cuando esta cargado', () => {
    const skeleton = renderer.create(<Skeleton isLoaded={true} />).toJSON();
    expect(skeleton).toHaveStyleRule('width', '100%');
    expect(skeleton).toHaveStyleRule('height', 'auto');
    expect(skeleton).toHaveStyleRule('margin-top', '5px');
    expect(skeleton).toHaveStyleRule('margin-right', '5px');
    expect(skeleton).toHaveStyleRule('margin-bottom', '5px');
    expect(skeleton).toHaveStyleRule('margin-left', '5px');
  });
});
