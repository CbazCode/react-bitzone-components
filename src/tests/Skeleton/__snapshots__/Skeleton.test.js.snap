// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test en <Skeleton /> componente <Skeleton /> debe renderizarse correctamente 1`] = `
.c0 {
  width: 100%;
  height: auto;
  margin-top: 5px;
  margin-right: 5px;
  margin-bottom: 5px;
  margin-left: 5px;
  background-color: #cacaca;
  border-radius: 4px;
  -webkit-animation: dKHzAH 3s linear infinite;
  animation: dKHzAH 3s linear infinite;
  color: transparent;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

<div
  class="c0"
/>
`;
